tests/trainers/README.md

```diff
@@ -0,0 +1,100 @@
+BCI tests modules
+=================
+
+Description
+-----------
+
+This directory contains modules and files that shall be copied into the container under test, for the execution of specific exercises and examples during the BCI validation .
+
+The copy from the test server to the container can be done in two main ways:
+
+
+1. Copy at build time, creating, in the test\_testname.py header, the content of a proper Dockerfile as a string, including COPY and any other command needed to the container preparation
+
+:..
+
+    DIR = "directory"
+    FILE = "filename.py"
+    CDIR = "container-dir"
+
+    DOCKERFIL = f"""
+    ...
+    COPY {DIR}/{FILE}  {CDIR}
+    ...
+    """
+
+    CONTAINER1 = pytest.param(
+        DerivedContainer(
+            base=container_from_pytest_param(CONTAINER),
+            containerfile=DOCKERFIL,
+        ),
+        marks=CONTAINER.marks,
+    )
+
+    CONTAINER_IMAGES_NEW = [ CONTAINER1, ... ]
+
+
+Then we parametrize the new container for each test and assign it to the auto_container_per_test fixtures, adding this last as input parameter of the test;
+
+:..
+
+    @pytest.mark.parametrize("auto_container_per_test", [CONTAINER_IMAGES_NEW], indirect=True)
+    def test_ testname(auto_container_per_test, ...):
+        ...
+
+
+In the test code we can run commands in the new container using the testinfra commands (See <https://testinfra.readthedocs.io/en/latest/modules.html>).
+
+I.e.:
+
+:..
+
+    auto_container_per_test.connection.run_expect([0], "python3 filename.py")
+
+
+2. Copy local files into the running container, using the 'podman|docker' cp command, via testinfra:
+
+:..
+
+    def test_<nnn>(auto_container_per_test, host, container_runtime):
+
+        CID = auto_container_per_test.container_id
+
+        host.run_expect( [0],
+            f"{container_runtime.runner_binary} cp {DIR}/{FILE}  {CID}:{CDIR}")
+
+
+Tests for BCI python
```

Comment:
This is a really great description!

Reply:
Thank you! 
Indeed in this description the _auto_container_per_test_ parameter could also be simply _container_.